File: training-data/ocaml/record_demo.ml
type point = { x : float; y : float }

let add_points a b = { x = a.x +. b.x; y = a.y +. b.y }

let () =
  let p1 = { x = 1.0; y = 2.0 } in
  let p2 = { x = 3.0; y = 4.0 } in
  let p3 = add_points p1 p2 in
  Printf.printf "(%.1f, %.1f)\n" p3.x p3.y

File: training-data/erlang/record_demo.erl
-module(record_demo).
-export([add_points/2]).

-record(point, {x, y}).

add_points(#point{x = X1, y = Y1}, #point{x = X2, y = Y2}) ->
    #point{x = X1 + X2, y = Y1 + Y2}.

main() ->
    P1 = #point{x = 1, y = 2},
    P2 = #point{x = 3, y = 4},
    io:format("~p~n", [add_points(P1, P2)]).
